chronicles.nim

```diff
@@ -216,14 +216,18 @@ macro expandIt*(T: type, expandedProps: untyped): untyped =
   when defined(debugLogImpl):
     echo result.repr
 
+template chroniclesUsedMagic(x: untyped) =
+  # Force the compiler to mark any symbol in the x
+  # as used without actually generate any code.
+  when compiles(x): discard
```

Comment:
can't they be annotated with `{.used.}` instead? this feels like it will further add to compilation slowness..

Reply:
Not always, for example:

```Nim
try:
   ...
except SomeError as exc:   # cannot use {.used.} here
   debug "Some text", msg=exc.msg
```

I don't know when we will get this feature in next nim version. Until then this is the available workaround.

Reply:
is there an upstream bug?

Reply:
It simply not exists